src/components/core/PivotTable.tsx

```diff
@@ -665,42 +665,21 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
         this.updateDesiredHeight(this.state.execution.executionResult);
     };
 
-    public onMenuAggregationClick = ({
-        type,
-        measureIdentifiers,
-        attributeIdentifier,
-        include,
-    }: IMenuAggregationClickConfig) => {
-        const columnTotals = this.getColumnTotals();
-
-        const columnTotalsChanged: AFM.ITotalItem[] = [];
-        for (const measureIdentifier of measureIdentifiers) {
-            columnTotalsChanged.push({ type, measureIdentifier, attributeIdentifier });
-        }
-
-        let updatedColumnTotals = [];
-        if (include) {
-            const columnTotalsChangedUnique = columnTotalsChanged.filter(
-                totalChanged => !columnTotals.some(total => isEqual(total, totalChanged)),
-            );
-
-            updatedColumnTotals = [...columnTotals, ...columnTotalsChangedUnique];
-        } else {
-            updatedColumnTotals = columnTotals.filter(
-                total => !columnTotalsChanged.find(totalChanged => isEqual(totalChanged, total)),
-            );
-        }
-
-        const newColumnTotals = sortBy(updatedColumnTotals, total => {
-            return renderedTotalTypesOrder.findIndex((rankedItem: string) => rankedItem === total.type);
-        });
+    public onMenuAggregationClick = (menuAggregationClickConfig: IMenuAggregationClickConfig) => {
+        const newColumnTotals = getUpdatedColumnTotals(
+            this.getColumnTotals(),
+            this.props.resultSpec,
+            menuAggregationClickConfig,
+        );
 
         this.props.pushData({
             properties: {
                 totals: newColumnTotals,
             },
         });
         this.setState({ columnTotals: newColumnTotals });
+
+        this.setGroupingProvider(this.props.groupRows && this.state.sortedByFirstAttribute);
```

Comment:
Called twice with same arguments > please move to some function

Reply:
Done